cmd/serverless-init/initcontainer/initcontainer.go

```diff
@@ -132,7 +153,9 @@ func flush(flushTimeout time.Duration, metricAgent serverless.FlushableAgent, tr
 	childCtx, cancel := context.WithTimeout(context.Background(), flushTimeout)
 	defer cancel()
 	go func(wg *sync.WaitGroup, ctx context.Context) {
-		logs.Flush(ctx)
+		if logsAgent != nil {
```

Comment:
Is it expected that this can be nil? If not, perhaps something should happen (log or even crash maybe) to indicate that there's a bug.

Reply:
It is possible and expected for this to sometimes be nil (logs disabled, error starting, etc). [This null check existed previously and used to be hidden under the global logs agent reference](https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/logs.go#L123), which I removed to be more explicit. 

There will be a log message if the agent fails to start. 

I am hesitant to add logging her in case the customer has logs intentionally disabled - we don't want to spam the agent logs with with expected failed flushes. But let me know if you disagree.

Reply:
I think using an optional makes more sense when the lifecycle is managed by FX, and in this case it's not. 
In the case of serverless, the logs agent instance is always non-nil so long as it does not fail to start. whereas in the core agent, the logs agent is optional because it can be disabled by config, and a failed `start` is fatal and will stop the agent. 

Once we move serverless to FX, I would be in favor of making this change to be consistent. Let me know if you disagree :)